src/middleware/batch.js

```diff
@@ -174,9 +174,16 @@ function sendRequests(requestMap: BatchRequestMap, next, opts) {
           throw new Error('Wrong response from server');
         }
 
-        batchResponse.payload.forEach(res => {
+        const responseHasIds: boolean = batchResponse.payload.every(response => response.id);
+        if (!responseHasIds && ids.length !== batchResponse.payload.length) {
```

Comment:
This check could be worthwile to have in any case - I guess the number of responses should always be the same as the number of requests?

Reply:
Who knows. Any server may have bugs, and we should catch this somehow and reject with error by spec.